code/addons/essentials/src/docs/preset.ts

```diff
@@ -1,8 +1,13 @@
-/* eslint-disable import/export */
+import { dirname, join } from 'path';
+
+// eslint-disable-next-line import/export
 export * from '@storybook/addon-docs/dist/preset';
 
 export const mdxLoaderOptions = async (config: any) => {
   // eslint-disable-next-line no-param-reassign
-  config.mdxCompileOptions.providerImportSource = '@storybook/addon-essentials/docs/mdx-react-shim';
+  config.mdxCompileOptions.providerImportSource = join(
+    dirname(require.resolve('@storybook/addon-docs/package.json')),
+    '/dist/shims/mdx-react-shim'
```

Comment:
Why do we need to use `/dist/shims` here, when the [exports](https://github.com/storybookjs/storybook/blob/f83212b5ae347caa0893a37e1201c70dcd531387/code/addons/docs/package.json#L62) of `addon-docs` includes `mdx-react-shim` ?  I thought the way we normally handle bundlers that don't support `exports` was by putting a file at the root which re-exported from `dist`.  That feels like a cleaner solution to me, rather than sprinkling `dist` throughout our source.

Reply:
@IanVS right, I'd definitely not want users to import directly from `dist`, because what's in `dist` is internal for us.
But that's also the reaosn why I think here it's OK to use:
This is all contained in the monorepo, so I'm less afraid this will break.

Sure I could direct to a root file, and then the root file directs to dist.. but I'd rather have less hoops and less files that need to be there for obscure reasons.

If this works, and therefore 1 less reason for having an root files that's 👍 for me.

I'm hoping that long term we won't have any root files anymore.. but that's likely going to be around the same time when:
> we want to support true node.js ESM, we'll need to start using createRequire, since our .mjs files that are generated right now will throw errors if require is not available, but that can be a matter for another day.